public/components/common/seach/autocomplete.tsx

```diff
@@ -0,0 +1,270 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+import './search.scss';
+import React, { createElement, Fragment, useEffect, useRef, useState } from 'react';
+import { render } from 'react-dom';
+import { h } from 'preact';
```

Comment:
what's preact

Reply:
Do we need preact? I thought dangerouslySetInnerHTML is good enough to add html.

Reply:
Nope, we don't need it. Will remove